src/java.base/share/classes/java/net/Socket.java

```diff
@@ -650,12 +658,17 @@ public void connect(SocketAddress endpoint) throws IOException {
      *        cause it to wakeup and close the socket. This method will then throw
      *        {@code SocketException} with the interrupt status set.
      * </ol>
+     * <p> Besides above noted cases, the socket will also be closed when a
+     * connection attempt fails with an {@link IOException}.
      *
      * @param   endpoint the {@code SocketAddress}
      * @param   timeout  the timeout value to be used in milliseconds.
      * @throws  IOException if an error occurs during the connection, the socket
      *          is already connected or the socket is closed
-     * @throws  SocketTimeoutException if timeout expires before connecting
+     * @throws  SocketTimeoutException if timeout expires before connecting,
+     *          the socket is closed
```

Comment:
The updated method description makes it clear that the the socket will be closed if the timeout expires before the connection is established, shouldn't need to change the exception description.

Reply:
Fixed in 424fc8ccd3a495a49f859f0cfc563cf463b3efeb.